severity.go

```diff
@@ -0,0 +1,51 @@
+package log
+
+import (
+	"fmt"
+	"strings"
+	"sync/atomic"
+)
+
+type Severity int32
+
+const (
+	SeverityInfo Severity = iota
+	SeverityWarning
+	SeverityError
+)
+
+var severityName = map[Severity]string{
+	SeverityInfo:    "INFO",
+	SeverityWarning: "WARN",
+	SeverityError:   "ERROR",
+}
+
+func (s *Severity) Get() Severity {
+	return Severity(atomic.LoadInt32((*int32)(s)))
+}
+
+func (s *Severity) Set(val Severity) {
+	atomic.StoreInt32((*int32)(s), int32(val))
+}
+
+func (s *Severity) Gte(val Severity) bool {
+	return s.Get() >= val
+}
+
+func (s Severity) String() string {
```

Comment:
I mentioned this in another comment, but I propose not exporting any of this. Then we can just guarantee that none of the calls to this function result in errors. And that is easy enough to do: just use `severityInfo`, `severityWarning` and `severityError` and we'll be fine.